File: Python/#121_maxProfit.py
class Solution:
    def maxProfit(self, prices: List[int]) -> int:
        if prices == []:
            return 0
        max_profit, min_price = 0, float('inf')
        for price in prices:
            min_price = min(price, min_price)
            profit = price - min_price
            max_profit = max(max_profit,profit)
        return max_profit

File: Java/#121_maxProfit.java
class Solution {
    public int maxProfit(int[] prices) {
        int min=Integer.MAX_VALUE, max=0;
        for(int p:prices){
            min=Math.min(min,p);
            int profit=p-min;
            max=Math.max(max,profit);
        }
        return max;
        
    }
}
